jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/TraceEnrichingFilter.java

```diff
@@ -32,47 +33,47 @@
 @Provider
 public final class TraceEnrichingFilter implements ContainerRequestFilter, ContainerResponseFilter {
 
+    // Handles incoming request
     @Override
     public void filter(ContainerRequestContext requestContext) throws IOException {
-        String operation = new StringBuilder(requestContext.getMethod())
-                .append(" /")
-                .append(requestContext.getUriInfo().getPath())
-                .toString();
-
-        String traceId = requestContext.getHeaderString(Traces.HttpHeaders.TRACE_ID);
-        String parentSpanId = requestContext.getHeaderString(Traces.HttpHeaders.PARENT_SPAN_ID);
-        String spanId = requestContext.getHeaderString(Traces.HttpHeaders.SPAN_ID);
+        String operation = requestContext.getMethod() + " /" + requestContext.getUriInfo().getPath();
+        // The following strings are all nullable
+        String traceId = requestContext.getHeaderString(TraceHttpHeaders.TRACE_ID);
+        String spanId = requestContext.getHeaderString(TraceHttpHeaders.SPAN_ID);
```

Comment:
i turn the sampling header into an optional because i pass it on; comfortable with the nulls within this function.